### PR TITLE
Init BindGen unit tests

### PR DESCRIPTION
**Pull request by Spacesai1or**
Adds unit tests for `removeDeploymentSalt` helper method

### CODE REVIEW

**Comment by Spacesai1or** [comment]
Current dependencies on/for this PR:
* `develop`
  * **PR #8279** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/8279?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #8280** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/8280?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #8281** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/8281?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #8282** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/8282?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #8283** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/8283?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #8284** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/8284?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
              * **PR #8285** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/8285?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                * **PR #8650** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/8650?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
                  * **PR #8651** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/8651?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/8650?utm_source=stack-comment).

**Comment by spacesailor24** [comment]
Current dependencies on/for this PR:
* `develop`
  * **PR #8279** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/8279?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #8280** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/8280?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #8281** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/8281?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #8282** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/8282?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #8283** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/8283?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #8284** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/8284?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
              * **PR #8285** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/8285?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                * **PR #8650** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/8650?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
                  * **PR #8651** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/8651?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/8650?utm_source=stack-comment).

File: op-bindings/bindgen/remote_handlers_test.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+func TestRemoveDeploymentSaltFailures(t *testing.T) {
+	generator := bindGenGeneratorRemote{}
+
+	for _, tt := range removeDeploymentSaltTestsFailures {
+		t.Run(tt.name, func(t *testing.T) {
+			_, err := generator.removeDeploymentSalt(tt.deploymentData, tt.deploymentSalt)
+
+			if err == nil {
+				t.Errorf("Expected error: %s but didn't receive it", tt.expectedError)
+				return
```
Comment (hamdiallam) [inline op-bindings/bindgen/remote_handlers_test.go]: We commonly use the `stretchr/testify` testing library for assertions. If the errors should have equal strings you could also do `require.Equal(t, err.String(), t.expectedError.String())`
```suggestion
		t.Run(tt.name, func(t *testing.T) {
			_, err := generator.removeDeploymentSalt(tt.deploymentData, tt.deploymentSalt)
                         require.Error(t, err, "expected %s but didn't receive it", tt.expectedError)
```

File: op-bindings/bindgen/remote_handlers_test.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+			}
+
+			if !strings.Contains(err.Error(), tt.expectedError) {
```
Comment (hamdiallam) [inline op-bindings/bindgen/remote_handlers_test.go]: can just use the above comment to flatten this all out into a single `require` statement

File: op-bindings/bindgen/remote_handlers_test.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+func TestRemoveDeploymentSalt(t *testing.T) {
+	generator := bindGenGeneratorRemote{}
+
+	for _, tt := range removeDeploymentSaltTests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, _ := generator.removeDeploymentSalt(tt.deploymentData, tt.deploymentSalt)
+			if diff := cmp.Diff(tt.expected, got); diff != "" {
+				t.Errorf("%s mismatch (-want +got):\n%s", tt.name, diff)
+			}
```
Comment (hamdiallam) [inline op-bindings/bindgen/remote_handlers_test.go]: require.Empty(t, cmp.Diff(tt.expected, got))

**Review by hamdiallam — APPROVED** [review]
Looks good! Can you add some comments about where the test cases come from and how the reader can reproduce them?

**Review by mds1 — APPROVED** [review]

**Pull request merged**